relayer/src/link/relay_path.rs

```diff
@@ -1227,45 +1227,55 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> RelayPath<ChainA, ChainB> {
         Ok(())
     }
 
-    pub fn process_pending_txs(&self) -> RelaySummary {
+    pub fn process_pending_txs(&self, clear_interval: u64) -> RelaySummary {
         if !self.confirm_txes {
             return RelaySummary::empty();
         }
 
-        let mut summary_src = self.process_pending_txs_src().unwrap_or_else(|e| {
-            error!("error processing pending events in source chain: {}", e);
-            RelaySummary::empty()
-        });
+        let mut summary_src = self
+            .process_pending_txs_src(clear_interval)
```

Comment:
Hmm, I'm wondering if the `clear_interval` name here conveys what's happening.

We could try something that is more explicit, like


```suggestion
        // If packet clearing is enabled (`clear_interval > 0`), then Hermes
        // will periodically clear packets as part of that mechanism.
        // Perform packets resubmission only when packet clearing is disabled,
        // i.e., when `clear_interval == 0`).
        let do_resubmit = if clear_interval == 0 { true } else { false };
        let mut summary_src = self
            .process_pending_txs_src(clear_interval)
```

This will avoid checking `if clear_interval == 0` twice (within `process_pending_txs_dst` and `process_pending_txs_src`) and instead we'll do it once only in this snippet.